sdcm/cluster_azure.py

```diff
@@ -84,8 +87,24 @@ def is_spot(self):
     def check_spot_termination(self):
         """Check if a spot instance termination was initiated by the cloud.
         """
-        # todo: fix it
-        return False
+        try:
+            self.wait_ssh_up(verbose=False)
+            result = self.remoter.run(
+                'curl http://169.254.169.254/metadata/scheduledevents?api-version=2020-07-01 -H "Metadata: true"', verbose=False)
+            status = json.loads(result.stdout.strip())
+            if status["DocumentIncarnation"] == self.scheduled_events_document_incarnation:
+                return SPOT_TERMINATION_CHECK_OVERHEAD
+            for event in status["Events"]:
+                self.scheduled_events_document_incarnation = status["DocumentIncarnation"]
+                if event["EventType"] == "Preempt":
+                    message = f"got spot termination event for node: {event['Resources']}. VM will be evicted at: {event['NotBefore']} "
```

Comment:
Nitpick: the `will` is incorrect if the `event['NotBefore']` will be in the past from current read time, so the message can be improved, something like:
`VM eviction time is {event['NotBefore']}`